src/overlays/actors/ovl_En_Siofuki/z_en_siofuki.c

```diff
@@ -21,27 +30,286 @@ const ActorInit En_Siofuki_InitVars = {
+        this->unk_180 = -45.0f;
+        this->unk_17C = -45.0f;
+        this->actionFunc = func_80AFC544;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Siofuki/func_80AFBE8C.s")
+void func_80AFC478(EnSiofuki* this, GlobalContext* globalCtx) {
+    func_80AFBDC8(this, globalCtx);
+    func_80AFBE8C(this, globalCtx);
+    func_80AFC1D0(this, globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Siofuki/func_80AFC1D0.s")
+    if (((u16)this->dyna.actor.params >> 0xC & 0xF) == 1) {
+        if (Flags_GetSwitch(globalCtx, ((u16)this->dyna.actor.params >> 6) & 0x3F)) {
+            this->timer = 20;
+            this->actionFunc = func_80AFC3C8;
+            func_800800F8(globalCtx, 0x1392, 0x28, &this->dyna.actor, 0);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Siofuki/func_80AFC218.s")
+        if (Flags_GetTreasure(globalCtx, (u16)this->dyna.actor.params & 0x3F)) {
+            this->unk_180 = -45.0f;
+            this->unk_17C = -45.0f;
+            this->actionFunc = func_80AFC544;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Siofuki/func_80AFC34C.s")
+void func_80AFC544(EnSiofuki* this, GlobalContext* globalCtx) {
+    func_80AFBDC8(this, globalCtx);
+    func_80AFC1D0(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Siofuki/func_80AFC3C8.s")
+void EnSiofuki_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnSiofuki* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Siofuki/func_80AFC478.s")
+    this->actionFunc(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Siofuki/func_80AFC544.s")
+void EnSiofuki_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnSiofuki* this = THIS;
+    u32 x;
+    u32 y;
+    u32 gameplayFrames = globalCtx->gameplayFrames;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Siofuki/EnSiofuki_Update.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_siofuki.c", 654);
+    func_80093D84(globalCtx->state.gfxCtx);
+    Matrix_Translate(0.0f, this->unk_170, 0.0f, MTXMODE_APPLY);
+    Matrix_Scale(1.0f, 1.0f, 1.0f, MTXMODE_APPLY);
+    gSPMatrix(oGfxCtx->polyXlu.p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_en_siofuki.c", 662),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    x = gameplayFrames * 15;
+    y = gameplayFrames * -15;
+    gSPSegment(oGfxCtx->polyXlu.p++, 0x08, Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, x, y, 64, 64, 1, x, y, 64, 64));
+    gSPDisplayList(oGfxCtx->polyXlu.p++, D_06000B70);
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_en_siofuki.c", 674);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Siofuki/EnSiofuki_Draw.s")
+    if (this->sfxFlags & 1) {
+        f32 temp;
+        switch (((u16)thisx->params >> 0xC) & 0xF) {
+            case 0:
+                temp = (this->unk_180 - 10.0f) / 390.0f;
```

Comment:
It's not necessary, but this could be written as 
```suggestion
                temp = (this->unk_180 - 10.0f) / (400.0f - 10.0f);
```
to parallel the other case.